### PR TITLE
context: add open_device_with_fd()

### PR DESCRIPTION
**Pull request by elmarco**
See API documentation for details.

### CODE REVIEW

**Comment by a1ien** [comment]
I think this function must be unsafe. And also #34 
How I said I don't like this api, but if you describe why it's worth to implement I can add

**Comment by elmarco** [comment]
Agree, the function should probably be unsafe, similar to other FromRawFd Rust APIs.

Regarding ownership, we can document it that the DeviceHandle will own and close the fd on close/Drop.

It is a useful function, not only for Android, but on Linux. You can write a privileged process that open the USB device and give it to a user (spice-gtk had to facl the file to do that in the past, but we can do better now, this has several advantages over the old facl approach) Thanks

File: src/context.rs [hunk 1]
```diff
@@ -98,6 +101,28 @@ pub trait UsbContext: Clone + Sized + Send + Sync {
         Some(unsafe { DeviceHandle::from_libusb(self.clone(), ptr) })
     }
 
+    /// Opens the device with a pre-opened file descriptor.
+    ///
+    /// This is UNIX-only and platform-specific. It is currently working with
+    /// Linux/Android, but might work with other systems in the future.
+    ///
+    /// This function consumes ownership of the specified file descriptor. The
+    /// device handle will take responsibility for closing it.
```
Comment (a1ien) [inline src/context.rs:110]: No. This function not consume and device handle not closing it. [libusb_wrap_sys_device](https://libusb.sourceforge.io/api-1.0/group__libusb__dev.html#ga98f783e115ceff4eaf88a60e6439563c)

>The system device handle must remain open until libusb_close() is called. The system device handle will not be closed by libusb_close().

**Review by a1ien — CHANGES_REQUESTED** [review]

File: src/context.rs [hunk 2]
```diff
@@ -98,6 +101,28 @@ pub trait UsbContext: Clone + Sized + Send + Sync {
...
+    #[cfg(unix)]
+    unsafe fn open_device_with_fd(&self, fd: RawFd) -> crate::Result<DeviceHandle<Self>> {
```
Comment (a1ien) [inline src/context.rs:112]: Also doc alias libusb_wrap_sys_device.

File: src/context.rs (hunk 1 above)
Comment (elmarco) [reply src/context.rs:110]: Oops! Good catch. I wonder if we shouldn't do it then. What do you think?

Comment (a1ien) [reply src/context.rs:110]: Hmm... in theory we can create some wrapper struc which take raw handle and implement deref to DeviceHandle. But it's may be not very useful.

Comment (elmarco) [reply src/context.rs:110]: That's an option, or simply add a Option<RawFd> on DeviceHandle.

File: src/context.rs (hunk 2 above)
Comment (elmarco) [reply src/context.rs:113]: done

File: src/context.rs (hunk 1 above)
Comment (elmarco) [reply src/context.rs]: I dropped that approach, it will be caller responsabiity until we offer some other facilites.

**Comment by a1ien** [comment]
LGTM. Thanks

**Comment by elmarco** [comment]
Thanks!

**Pull request merged**